src/graph/validator/ACLValidator.cpp

```diff
@@ -104,10 +104,18 @@ Status ChangePasswordValidator::toPlan() {
 // grant role
 Status GrantRoleValidator::validateImpl() {
   const auto *sentence = static_cast<const GrantSentence *>(sentence_);
-  if (sentence->getAccount()->size() > kUsernameMaxLength) {
+  const std::string *account = sentence->getAccount();
+  if (account->size() > kUsernameMaxLength) {
     return Status::SemanticError("Username exceed maximum length %ld characters.",
                                  kUsernameMaxLength);
   }
+  auto metaClient = qctx_->getMetaClient();
```

Comment:
You could get roles from session directly.

Reply:
Sorry, it's my mistake, the granted user is `Account` from sentence and is different from current user of session.